Optionally build with source dependencies for akka

jrudolph opened the pull request.

ktoso commented on build.sbt:
```diff
@@ -1,14 +1,21 @@
+  sys.props += "akka.build.aggregateSamples" -> "false"
+
+  uri("git://github.com/akka/akka.git#release-2.4")
+}
```
Nice!

2m commented:
Great stuff. This reminded me of https://github.com/sbt/sbt/pull/2844

We could implement it similarly: if there is `-Dakka.path` set, depend on sources.

ktoso commented:
Yeah, we'd need it like that I believe.

jrudolph commented:
> We could implement it similarly: if there is -Dakka.path set, depend on sources.

I now called the environment variable `akka.sources` which needs to be a URI either to a git repo or a local file. I removed the redundant boolean setting to make the switch.

I already fixed two tests that wouldn't compile against current akka master anymore.

ktoso commented on project/Dependencies.scala:
```diff
@@ -113,39 +107,27 @@ object Dependencies {
   )
 
-  lazy val httpTests = l ++= Seq(Test.akkaMmltinodeTestKit, Test.junit, Test.scalatest.value, Test.junitIntf)
+  lazy val httpTests = l ++= Seq(Test.junit, Test.scalatest.value, Test.junitIntf)
```
multinode is used for the `perfspec` though AFAIR?

ktoso commented on project/AkkaDependency.scala:
```diff
@@ -0,0 +1,38 @@
+
+object AkkaDependency {
+  // Needs to be a URI like git://github.com/akka/akka.git#master or file:///xyz/akka
+  val akkaSourceDependencyUri = sys.props.getOrElse("akka.sources", "")
```
excellent 👍

ktoso commented on akka-http-core/src/test/scala/akka/http/impl/engine/ws/WSClientAutobahnTest.scala:
```diff
@@ -167,28 +164,8 @@ object WSClientAutobahnTest extends App {
-      (stage, p.future)
-    }
+  def completionSignal[T]: Flow[T, T, Future[Done]] =
+    Flow[T].watchTermination()((_, res) ⇒ res)
```
👍

ktoso commented on build.sbt:
```diff
@@ -88,6 +93,7 @@ lazy val httpTests = project("akka-http-tests")
   .enablePlugins(MultiNode)
   .disablePlugins(MimaPlugin) // this is only tests
   .configs(MultiJvm)
+  .addAkkaModuleDependency("akka-multi-node-testkit", "test")
```
oh I see, okey

ktoso commented:
LGTM, nice one!

jrudolph commented:
Thanks for merging. I'll setup jobs to run the builds nightly.
